Tensile/KernelWriterAssembly.py

```diff
@@ -7409,6 +7473,140 @@ def shiftVectorComponents(self, kernel, tP):
+    numSubOutputPerWave0       = numOutputThreads0 * numContinuousOutput
+    numSubOutputGroupsPerWave0 = kernel[tP["mt"]] // numSubOutputPerWave0
+    numShiftBlock              = numContinuousOutput // glvw
+    numOutputElements          = numSubOutputGroupsPerWave0 * numContinuousOutput
+
+    subTile1 = (kernel["MacroTile1"] // ((kernel["SubGroup0"] * kernel["SubGroup1"]) // numThreadInWave)) // kernel["MatrixInstN"]
+
+    # labels
+    svrLabels = []
+    sviLabels = []
+    svoLabels = []
+    for i in range(0, glvw):
+      r = (i+1) % glvw
+      label = self.getLabelNum("ShiftVectorComponents%u_R%u" % (tP["idx"], r) )
+      svrLabels.append(label)
+      tmpLabels = []
+      tmp2Labels = []
+      for v in range(0, numSubOutputGroupsPerWave0):
+        label = self.getLabelNum("ShiftVectorComponents%u_R%u_V%u" % (tP["idx"], r, v) )
+        tmpLabels.append(label)
+        tmp2Labels2 = []
+        for o in range(0, numShiftBlock):
+          label = self.getLabelNum("ShiftVectorComponents%u_R%u_V%u_O%u" % (tP["idx"], r, v, o) )
+          tmp2Labels2.append(label)
+        tmp2Labels.append(tmp2Labels2)
+      sviLabels.append(tmpLabels)
+      svoLabels.append(tmp2Labels)
+
+    # wgMT value
+    tmpSgpr = self.getTmpSgpr(2)
+    tmpVgpr = self.vgprPool.checkOut(2)
+    wgMT    = self.vgprPool.checkOut(1)
+
+    kStr += inst("v_mov_b32"    , vgpr(wgMT), sgpr(tP["wg"]), "")
+    kStr += inst("v_mul_i32_i24", vgpr(wgMT), hex(-kernel[tP["mt"]]), vgpr(wgMT), "wg*MT")
+    kStr += inst("_v_add_co_u32", vgpr(wgMT), "vcc", sgpr("SizesFree+%u"%tP["idx"]), vgpr(wgMT), "wgMT = Size - wg*MT")
+    kStr += inst("v_mov_b32"    , vgpr(tmpVgpr), hex(kernel[tP["mt"]]), "MT")
+    kStr += inst("v_cmp_lt_u32" , sgpr(tmpSgpr,2), vgpr(wgMT), vgpr(tmpVgpr), "wgMT < MT" )
+    kStr += inst("v_cndmask_b32", vgpr(wgMT), vgpr(tmpVgpr), vgpr(wgMT), sgpr(tmpSgpr,2), "wgMT = (wgMT < MT) ? wgMT : MT" )
+    dummy = self.vgprPool.checkOut(1)
+
+    # rReg
+    rReg = self.vgprPool.checkOut(1)
+    divisor = glvw
+    kStr += vectorStaticRemainder(dummy, rReg, wgMT, divisor, tmpVgpr, tmpSgpr)
+
+    # gReg
+    gReg = self.vgprPool.checkOut(1)
+    divisor = numSubOutputPerWave0  # glvw
+    kStr += vectorStaticDivide(gReg, wgMT, divisor, tmpVgpr, tmpSgpr)
+
+    # eReg
+    eReg = self.vgprPool.checkOut(1)
+    divisor = numContinuousOutput
+    kStr += vectorStaticRemainder(dummy, eReg, wgMT, divisor, tmpVgpr, tmpSgpr)
+
+    mReg = self.vgprPool.checkOut(1)
+    divisor = numContinuousOutput
+    kStr += vectorStaticDivide(mReg, wgMT, divisor, tmpVgpr, tmpSgpr)
+    divisor = numOutputThreads0
+    kStr += vectorStaticRemainder(dummy, mReg, mReg, divisor, tmpVgpr, tmpSgpr)
+
+    tReg = self.vgprPool.checkOut(1)
+    divisor = numThreadInWave
+    kStr += vectorStaticRemainder(dummy, tReg, "Serial", divisor, tmpVgpr, tmpSgpr)
+    divisor = kernel["MatrixInstN"]
+    kStr += vectorStaticDivide(tReg, tReg, divisor, tmpVgpr, tmpSgpr)
```

Comment:
Comments would help out here

Reply:
update comment and make a slide to explain in e-mail